torchani/models.py

```diff
@@ -43,6 +56,77 @@ def forward(self, species_coordinates):
         species_energies = self.neural_networks(species_aevs)
         return self.energy_shifter(species_energies)
 
+    def __getitem__(self, index):
+        ret = torch.nn.Sequential(self.aev_computer,
+                                  self.neural_networks[index],
+                                  self.energy_shifter)
+
+        def ase(**kwargs):
+            from . import ase
+            return ase.Calculator(self.builtins.species, self.aev_computer,
+                                  self.neural_networks[index],
+                                  self.energy_shifter, **kwargs)
+
+        ret.ase = ase
+        ret.species_to_tensor = self.builtins.consts.species_to_tensor
+        return ret
+
+    def __len__(self):
+        return len(self.neural_networks)
+
+    def ase(self, **kwargs):
+        """Get an ASE Calculator using this model"""
+        from . import ase
+        return ase.Calculator(self.builtins.species, self.aev_computer,
+                              self.neural_networks, self.energy_shifter,
+                              **kwargs)
+
+    def species_to_tensor(self, *args, **kwargs):
+        """Convert species from strings to tensor.
+
+        See also :method:`torchani.neurochem.Constant.species_to_tensor`"""
+        return self.builtins.consts.species_to_tensor(*args, **kwargs) \
+            .to(self.aev_computer.ShfR.device)
+
+
+class BuiltinNet(torch.nn.Module):
```

Comment:
Why creating a `BuiltinNet`? Wouldn't it be better if we just keep `BuiltinModels` and move the implementation of `BuiltinNet` into `BuiltinModels`? `BuiltinModels` is internal, it is not in the documents, so it is safe to change this.

Reply:
BuiltinModels takes as an input a BuiltinsAbstract object and then gets all the important attributes from that object. BuiltinNet takes the paths as an input directly. I'm afraid that someone has been using BuiltinModels and since they take different inputs changing this will break their code. 

I know its not in the documentation but since you were worried about people's code breaking I thought this would be safer.

One alternative is for BuiltinModels to take the file paths and a BuiltinsAbstract object optionally, and add a deprecation warning if people input the BuiltinsAbstract object, but I'm afraid that is going to make the code more confusing.

If you prefer though, I can delete BuiltinModels and rename BuiltinNet to BuiltinModels. At some point BuiltinModels needs to be deleted anyways. Perhaps the name is not the greatest thing now that I think of it, it should be plural right, BuiltinNets?

Reply:
Plural or not is not a big deal, either are OK to me. I just don't like the word `Net`, but it is not a big deal either. We can leave this part as is.